lib/astutils.cpp

```diff
@@ -204,6 +204,9 @@ static const Token * followVariableExpression(const Token * tok, bool cpp)
         if (const Variable * var2 = tok2->variable()) {
             if(!var2->scope())
                 return tok;
+            // If type is `case` then we haven't found the correct variable declaration
+            if(Token::simpleMatch(var2->typeStartToken(), "case"))
```

Comment:
Ideally the `typeStartToken` should not return `case`. It should return `a`.

Ideally you would fix the `SymbolDatabase` instead because then you fix the real issue. But I am fine with this quick fix also. But then it needs to be clarified in a comment.. something like:

    // FIXME This is a quick fix. Fix SymbolDatabase so typeStartToken points at the variable declaration.

I am hoping that we can solve this properly someday.

Reply:
Yea, I am not as familiar with the SymbolDatabase code, but I will add the comment to make it clearer.